In [11]:
# Import Libraries
from skimage.io import imshow, imread
from skimage.transform import rescale, resize
import os
import numpy as np
from PIL import ImageFile
ImageFile.LOAD_TRUNCATED_IMAGES = True
import tensorflow.compat.v1 as tf
tf.disable_v2_behavior()

In [1]:
# This code is initially used to generate X.npy and Y.npy

# samples = []
# truths = []
# for folder in os.listdir("HTLID"):
#     for subfolder in os.listdir("HTLID/{}".format(folder)):
#         i = 0
#         for name in os.listdir("HTLID/{}/{}".format(folder, subfolder)):
#             samples.append("HTLID/" + folder + "/" + subfolder + "/" + name)
#             if name[len(name)-9:len(name)] != "_mask.png":
#                 if name[0:2] == 'co':
#                     truths.append(48)
#                 elif name[0:2] == 'da':
#                     truths.append(46)
#                 elif name[0:2] == 'ex':
#                     truths.append(0)
#                 elif name[0:2] == 'fl':
#                     truths.append(45)
#                 elif name[0:2] == 'no':
#                     truths.append(43)
#                 elif name[0:2] == 'or':
#                     truths.append(50)
#                 elif name[0:2] == 'ox':
#                     truths.append(49)
#                 else:
#                     truths.append(55)

# print(len(samples))
# masks=[]
# i = 0
# while i < len(samples):
#     if samples[i][len(samples[i])-9:len(samples[i])] == "_mask.png":
#         masks.append(samples[i])
#         samples.remove(samples[i])
#         i -= 1
#     i += 1
# Y = keras.utils.to_categorical(truths, num_classes = 63)

In [4]:
Y=np.load("Y.npy")

In [5]:
# X=[]
# I = np.linspace(0, 3455, 3456, dtype=np.float64).reshape(3456,1)
# J = np.linspace(0, 5183, 5184, dtype=np.float64).reshape(1, 5184)
# for l in range(0, 600):
#     img = imread(masks[l])
#     actual = imread(masks[l][0:len(masks[l])-9] + ".jpg")
#     i = int(np.sum(I*img)/np.sum(img))
#     j = int(np.sum(J*img)/np.sum(img))
#     i_max = np.max(np.argmax(img*I, axis=0))
#     j_max = np.max(np.argmax(img*J, axis=1))
#     _from = int(np.max((i_max - i, j_max - j)))
#     cut = actual[(i - _from):(i + _from),(j - _from):(j + _from)]
#     img_in = resize(cut, (128,128))
#     X.append(img_in)
#     print(l)

In [6]:
X = np.load("X.npy")

In [12]:
# Prevent Memory Hog
from keras.backend.tensorflow_backend import set_session
config = tf.ConfigProto()
config.gpu_options.allow_growth=True
set_session(tf.Session(config=config))

In [7]:
# Load model trained by Model2Training
from keras.models import load_model
model = load_model('one_pic7_1')

In [8]:
#
model.evaluate(X, Y)

600/600 [==============================] - 3s 5ms/step


[0.564058409134547, 0.8083333325386047]

This model has a loss of 0.56, and an accuracy of 80.8%